Mathlib/Algebra/SkewMonoidAlgebra/Basic.lean

```diff
@@ -160,6 +153,266 @@ theorem support_eq_empty {p} : p.support = ∅ ↔ (p : SkewMonoidAlgebra k G) =
+  f.toFinsupp.sum g
+
+theorem sum_def {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    sum f g = ∑ a ∈ f.support, g a (f.coeff a) := rfl
+
+@[simp]
+theorem sum_single_index {N} [AddCommMonoid N] {a : G} {b : k} {h : G → k → N}
+    (h_zero : h a 0 = 0) : (SkewMonoidAlgebra.single a b).sum h = h a b :=
+  Finsupp.sum_single_index h_zero
+
+theorem map_sum {N P : Type*} [AddCommMonoid N] [AddCommMonoid P] {H : Type*} [FunLike H N P]
+    [AddMonoidHomClass H N P] (h : H) (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    h (sum f g) = sum f fun (a : G) (b : k) => h (g a b) :=
+  _root_.map_sum h _ _
+
+theorem toFinsupp_sum {N : Type*} [AddCommMonoid N] (f : SkewMonoidAlgebra k G) (g : G → k → N) :
+    sum f g = Finsupp.sum f.toFinsupp g := rfl
+
+/-- Variant where the image of `g` is a `SkewMonoidAlgebra`. -/
+theorem toFinsupp_sum' {k' G' : Type*} [AddCommMonoid k'] (f : SkewMonoidAlgebra k G)
+    (g : G → k → SkewMonoidAlgebra k' G') :
+    (sum f g).toFinsupp = Finsupp.sum f.toFinsupp (fun (a : G) (b : k) ↦ (g a b).toFinsupp) :=
+  _root_.map_sum toFinsuppAddEquiv (fun (a : G) => g a (f.coeff a)) f.toFinsupp.support
+
+theorem ofFinsupp_sum {k' G' : Type*} [AddCommMonoid k'] (f : G →₀ k)
+    (g : G → k → G' →₀ k'):
+    (⟨Finsupp.sum f g⟩ : SkewMonoidAlgebra k' G') = sum ⟨f⟩ (fun (a : G) (b : k) ↦ ⟨(g a b)⟩) := by
+  apply toFinsupp_injective; simp only [toFinsupp_sum']
+
+theorem sum_single (f : SkewMonoidAlgebra k G) : f.sum single = f := by
+  apply toFinsupp_injective; simp only [toFinsupp_sum', toFinsupp_single, Finsupp.sum_single]
+
+/-- Taking the `sum` under `h` is an additive homomorphism, if `h` is an additive homomorphism.
+This is a more specific version of `SkewMonoidAlgebra.sum_add_index` with simpler hypotheses. -/
+theorem sum_add_index' {S : Type*} [AddCommMonoid S] {f g : SkewMonoidAlgebra k G} {h : G → k → S}
+    (hf : ∀ i, h i 0 = 0) (h_add : ∀ a b₁ b₂, h a (b₁ + b₂) = h a b₁ + h a b₂) :
+    (f + g).sum h = f.sum h + g.sum h := by
+  rw [show f + g = ⟨f.toFinsupp + g.toFinsupp⟩ by rw [ofFinsupp_add, eta]]
+  exact Finsupp.sum_add_index' hf h_add
+
+/-- Taking the `sum` under `h` is an additive homomorphism, if `h` is an additive homomorphism.
+This is a more general version of `SkewMonoidAlgebra.sum_add_index'`;
+the latter has simpler hypotheses. -/
+theorem sum_add_index {S : Type*} [DecidableEq G] [AddCommMonoid S]
+    {f g : SkewMonoidAlgebra k G} {h : G → k → S} (h_zero : ∀ a ∈ f.support ∪ g.support, h a 0 = 0)
+    (h_add : ∀ a ∈ f.support ∪ g.support, ∀ (b₁ b₂ : k), h a (b₁ + b₂) = h a b₁ + h a b₂) :
+    (f + g).sum h = f.sum h + g.sum h := by
+  rw [show f + g = ⟨f.toFinsupp + g.toFinsupp⟩ by rw [ofFinsupp_add, eta]]
+  exact Finsupp.sum_add_index h_zero h_add
+
+@[simp]
+theorem sum_add {S : Type*} [AddCommMonoid S] (p : SkewMonoidAlgebra k G) (f g : G → k → S) :
+    (p.sum fun n x => f n x + g n x) = p.sum f + p.sum g := Finsupp.sum_add
```

Comment:
Please use `↦` in new code added to mathlib (also below). You are using `↦` earlier in the file; maybe here you copied some old code and forgot to change it?
```suggestion
    (p.sum fun n x ↦ f n x + g n x) = p.sum f + p.sum g := Finsupp.sum_add
```